src/hnsw/build.c

```diff
@@ -161,19 +215,24 @@ static int GetArrayLengthFromHeap(Relation heap, int indexCol)
         return n_items;
     }
 
-    // Get the indexed column out of the row and return it's dimensions
-    datum = heap_getattr(tuple, indexCol, RelationGetDescr(heap), &isNull);
-    if(!isNull) {
-        array = DatumGetArrayTypePCopy(datum);
-        n_items = ArrayGetNItems(ARR_NDIM(array), ARR_DIMS(array));
+    if(indexInfo->ii_Expressions != NULL) {
+        Expr *indexpr_item = lfirst(list_head(indexInfo->ii_Expressions));
```

Comment:
Can this list have more expressions? if so, when?
Either case, can you write a comment and relevant invariant check? E.g. if the list will always have a single element, then add an invariant check asserting that the list length is 1

Reply:
It can not contain more than 1 element as our hnsw index does not support multi column indexing. Added invariant check for it